extensions/features-clustering/deployment/src/main/java/io/quarkiverse/cxf/features/clustering/deployment/QuarkusCxfClusteringProcessor.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkiverse.cxf.features.clustering.deployment;
+
+import java.util.stream.Stream;
+
+import org.apache.cxf.clustering.FailoverFeature;
+import org.apache.cxf.clustering.LoadDistributorFeature;
+import org.apache.cxf.clustering.circuitbreaker.CircuitBreakerFailoverFeature;
+import org.apache.cxf.clustering.circuitbreaker.ZestCircuitBreaker;
+
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+
+public class QuarkusCxfClusteringProcessor {
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem("cxf-rt-features-clustering");
+    }
+
+    @BuildStep
+    void indexDependencies(BuildProducer<IndexDependencyBuildItem> indexDependencies) {
+        Stream.of(
+                "org.apache.cxf:cxf-rt-features-clustering")
+                .forEach(ga -> {
+                    String[] coords = ga.split(":");
+                    indexDependencies.produce(new IndexDependencyBuildItem(coords[0], coords[1]));
+                });
+    }
+
+    @BuildStep
+    void unremovableBean(BuildProducer<UnremovableBeanBuildItem> unremovable) {
+        unremovable.produce(UnremovableBeanBuildItem.beanTypes(FailoverFeature.class,
+                LoadDistributorFeature.class, CircuitBreakerFailoverFeature.class,
+                ZestCircuitBreaker.class));
+    }
```

Comment:
I propose to add configuration with 
list of alternative address by strategy (random, failover,...) as a config like 
quarkus.cxf.client."failover-client".failoverAddresses=http://adress1,http://adress2
then generate feature on build time and add it automaticly. wdyt ?

Reply:
Thanks @dufoli , this is a good idea, however I'm not finding time to complete this PR ... there is just one outstanding item for an initial cut of the clustering feature integration ... basically the CircuitBreakerFailoverFeatureTest is not completed.  It's a bit more complex test as we'd need to start and stop the container endpoints during the test.

@ppalaga Are you ok with creating an issue for CircuitBreakerFailoverFeatureTest test being completed, while we merge in the rest of the PR?  Otherwise, it's also totally fine if @dufoli or you take this PR over to completion, as I am likely not going to be able to dedicate some time in the next week or two.

Reply:
as you want, you have done a great job on maintaining my baby project. So glad if I can take your PR or @ppalaga can do so. Anyway, I have take a look to the job done and I am happy to see how everythings was split to smaller sub module and class. I have never take the time to do so and as I said it was my first project in java so it was to discover a bit with help of my brother (java archi).

Reply:
sounds good @dufoli !!  Thank you!  Please feel free to take this one ... we can close this PR as needed, and submit a new one from your end!